docs/rules-from-web-poet.rst

```diff
@@ -187,21 +203,103 @@ Take note that since ``SCRAPY_POET_OVERRIDES`` is structured as
+returning for a given URL.
+
+Let's check out an example:
+
+.. code-block:: python
+
+    import attrs
+    import scrapy
+    from web_poet import WebPage, handle_urls, field
+    from scrapy_poet import DummyResponse
+
+
+    @attrs.define
+    class Product:
+        name: str
+
+
+    @handle_urls("example.com")
+    @attrs.define
+    class ProductPage(WebPage[Product]):
+
+        @field
+        def name(self) -> str:
+            return self.css("h1.name ::text").get("")
+
+
+    class MySpider(scrapy.Spider):
+        name = "myspider"
+
+        def start_requests(self):
+            yield scrapy.Request(
+                "https://example.com/products/some-product", self.parse
+            )
+
+        # We can directly ask for the item here instead of the page object.
+        def parse(self, response: DummyResponse, item: Product):
+            return item
+
+From this example, we can see that:
+
+    * Spider callbacks can directly ask for items as dependencies.
+    * The ``Product`` item instance directly comes from ``ProductPage``.
+    * This is made possible by the ``ApplyRule("example.com", use=ProductPage,
+      to_return=Product)`` instance created from the ``@handle_urls`` decorator
+      on ``ProductPage``.
+
+.. note::
+
+    The slightly longer alternative way to do this is by declaring the page
+    object itself as the dependency and then calling its ``.to_item()`` method.
+    For example:
+
+    .. code-block:: python
+
+        @handle_urls("example.com")
+        @attrs.define
+        class ProductPage(WebPage[Product]):
+            product_image_page: ProductImagePage
+
+            @field
+            def name(self) -> str:
+                return self.css("h1.name ::text").get("")
+
+            @field
+            async def image(self) -> Image:
+                return await self.product_image_page.to_item()
+
+
+        class MySpider(scrapy.Spider):
+            name = "myspider"
+
+            def start_requests(self):
+                yield scrapy.Request(
+                    "https://example.com/products/some-product", self.parse
+                )
+
+            async def parse(self, response: DummyResponse, product_page: ProductPage):
+                return await product_page.to_item()
+
+For more information about all the expected behavior for the ``to_return``
+parameter in :class:`web_poet.rules.ApplyRule`, check out web-poet's tutorial
+regarding :ref:`rules-item-class-example`.
+
+
+Registry
+========
+
+As mentioned above, scrapy-poet uses the registry from web-poet called
+:class:`web_poet.rules.RulesRegistry`.
+
+This registry implementation can be changed if needed. A different registry can
+be configured by passing its class path to the ``SCRAPY_POET_REGISTRY`` setting.
+Such registries must be a subclass of :class:`web_poet.rules.RulesRegistry`
+to ensure the expected methods and its types are properly accounted for.
```

Comment:
I don't think it's enough to describe the API. It's not clear which methods are needed, and which are implementation details. A few options:

1. Remove SCRAPY_POET_REGISTRY setting
2. Keep the setting, but remove the detailed notes about the registry ("Such registries must be a subclass of ..."); leave it semi-documented.
3. Provide an official API which users need to implement, probably as an abc class. Or, maybe, make sure that all methods which are not used directly are _underscored.

What do you think? I'm +0.5 to (2).

Reply:
Although (1) might be even better; see https://github.com/scrapinghub/scrapy-poet/pull/88/files#r1073900090

Reply:
As discussed in the call, it'd be probably best to remove the support for replacing the registry. https://github.com/scrapinghub/scrapy-poet/pull/88/commits/e3b7a8e03c284c605005394caf8e0b13af7bc8d5